src/pw/pw_methods.F

```diff
@@ -2661,4 +2685,54 @@ SUBROUTINE pw_set_value(pw, value)
 
    END SUBROUTINE pw_set_value
 
+! **************************************************************************************************
+!> \brief ...
+!> \param pw1 ...
+!> \param pw2 ...
+! **************************************************************************************************
+   SUBROUTINE pw_multiply_with(pw1, pw2)
+      TYPE(pw_type), INTENT(IN)                          :: pw1, pw2
```

Comment:
I am wondering, if the INTENT(IN) attribute for `pw1` is appropriate here, because it appears on the lhs.

Reply:
Thanks for considering my comment. Now, I am confused, because the dummy argument `pw1` has no `POINTER` attribute. I haven't studied the Fortran standard but only a common Fortran textbook which states:
![Screenshot 2023-08-23 at 16 01 19](https://github.com/cp2k/cp2k/assets/10218869/5e95edab-a9e9-41d7-a1c7-9e265c7844a5)
Well, no clue how Fortran compiler handle this eventually, but I am not sure, if we can rely on any analogy to C here.